src/test/java/io/logz/BaseTest.java

```diff
@@ -0,0 +1,147 @@
+        logzioLogbackAppender.setContext(logbackContext);
+        logzioLogbackAppender.setToken(token);
+        logzioLogbackAppender.setLogzioType(type);
+        logzioLogbackAppender.setDebug(true);
+        logzioLogbackAppender.setLogzioUrl("http://" + LISTENER_ADDRESS + ":" + port);
+        logzioLogbackAppender.setAddHostname(addHostname);
+
+        if (drainTimeout != null) {
+            logzioLogbackAppender.setDrainTimeoutSec(drainTimeout);
+        }
+        if (fsPercentThreshold != null) {
+            logzioLogbackAppender.setFileSystemFullPercentThreshold(fsPercentThreshold);
+        }
+        if (bufferDir != null) {
+            logzioLogbackAppender.setBufferDir(bufferDir);
+        } else {
+            File tempDir = TestEnvironment.createTempDirectory();
+            tempDir.deleteOnExit();
+            logzioLogbackAppender.setBufferDir(tempDir.getAbsolutePath());
+        }
+        if (socketTimeout != null) {
+            logzioLogbackAppender.setSocketTimeout(socketTimeout);
+        }
+
+        if (additionalFields != null) {
+            logzioLogbackAppender.setAdditionalFields(additionalFields);
+        }
+
+        if (gcPersistedQueueFilesIntervalSeconds != null) {
+            logzioLogbackAppender.setGcPersistedQueueFilesIntervalSeconds(gcPersistedQueueFilesIntervalSeconds);
+        }
+
+        logzioLogbackAppender.start();
+        assertThat(logzioLogbackAppender.isStarted()).isTrue();
+        logbackLogger.addAppender(logzioLogbackAppender);
+        logbackLogger.setAdditive(false);
+
+        return logbackLogger;
+    }
+
+    protected String random(int numberOfChars) {
+        return UUID.randomUUID().toString().substring(0, numberOfChars-1);
+    }
+
+    protected void assertNumberOfReceivedMsgs(int count) {
+        assertThat(mockListener.getNumberOfReceivedLogs())
+                .describedAs("Messages on mock listener: {}", mockListener.getReceivedMsgs())
+                .isEqualTo(count);
+    }
+
+    protected void assertLogReceivedIs(String message, String token, String type, String loggerName, Level level) {
+        MockLogzioBulkListener.LogRequest log = assertLogReceivedByMessage(message);
+        assertLogReceivedIs(log, token, type, loggerName, level);
+    }
+
+    protected void assertLogReceivedIs(MockLogzioBulkListener.LogRequest log, String token, String type, String loggerName, Level level) {
+        assertThat(log.getToken()).isEqualTo(token);
+        assertThat(log.getType()).isEqualTo(type);
+        assertThat(log.getLogger()).isEqualTo(loggerName);
+        assertThat(log.getLogLevel()).isEqualTo(level.toString());
+    }
+
+    protected MockLogzioBulkListener.LogRequest assertLogReceivedByMessage(String message) {
+        Optional<MockLogzioBulkListener.LogRequest> logRequest = mockListener.getLogByMessageField(message);
+        assertThat(logRequest.isPresent()).describedAs("Log with message '"+message+"' received").isTrue();
```

Comment:
get() is to take the actual LogRequest object. isPreset() simply says: Do we have a LogRequest